pkgs/build-support/buildenv/builder.pl

```diff
@@ -84,6 +84,10 @@ sub checkCollision {
 sub findFiles {
     my ($relName, $target, $baseName, $ignoreCollisions, $checkCollisionContents, $priority) = @_;
 
+    if (-f $target) {
+        die "Path $target is a file and can't be merged into an environment using pkgs.buildEnv!";
```

Comment:
...but should it not be possible?

Reply:
I'd say it shouldn't. If I understand the code right it never worked properly and merging a file with a directory feels wrong.

if you want to have a store path which contains the file from `writeText`, you may want to use `writeTextDir` instead.